File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03403/p03403_num7_parsed.py
# PROGRAM

N = int(input())
spots = list(map(int, input().split()))
total_cost = sum([abs(spots[i] - spots[i-1]) for i in range(1, N+1)])

for i in range(N):
    if i == 0:
        cost = abs(spots[i] - spots[i+1])
    elif i == N-1:
        cost = abs(spots[i] - spots[i-1])
    else:
        cost = abs(spots[i] - spots[i-1]) + abs(spots[i] - spots[i+1])
    print(total_cost - cost)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03403/p03403_num7_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int main() {
    int N;
    cin >> N;

    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }

    vector<long long> cost(N);
    cost[0] = 0;
    for (int i = 1; i < N; i++) {
        cost[i] = cost[i-1] + abs(A[i] - A[i-1]);
    }
    cost[N-1] += abs(A[N-1] - A[0]);

    for (int i = 0; i < N; i++) {
        long long total_cost = cost[N-1] - abs(A[i] - A[(i-1+N) % N]) - abs(A[i] - A[(i+1) % N]);
        cout << total_cost << endl;
    }

    return 0;
}